pkgs/development/python-modules/deap/default.nix

```
@@ -9,19 +18,18 @@ buildPythonPackage rec {
     hash = "sha256-zAHemJLfp9G8mAPasoiS/q0XfwGCyB20c2CiQOrXeP8=";
   };
 
-  propagatedBuildInputs = [ numpy matplotlib ];
+  propagatedBuildInputs = [ matplotlib numpy pytest ];
 
-  nativeCheckInputs = [ nose ];
   checkPhase = ''
-    nosetests --verbosity=3
+    runHook preCheck
+    ${python3.interpreter} -m unittest
+    runHook postCheck
   '';
 
   meta = with lib; {
     description = "DEAP is a novel evolutionary computation framework for rapid prototyping and testing of ideas.";
     homepage = "https://github.com/DEAP/deap";
     license = licenses.lgpl3;
```

```suggestion
    license = licenses.lgpl3Plus;
```
https://github.com/DEAP/deap/blob/ed7f04d330880839c11e4e624566c9fa616181bb/deap/__init__.py#L1-L6